subsys/mgmt/mcumgr/smp_bt.c

```diff
@@ -156,6 +159,63 @@ static ssize_t smp_bt_chr_write(struct bt_conn *conn,
 				const void *buf, uint16_t len, uint16_t offset,
 				uint8_t flags)
 {
+#ifdef CONFIG_MCUMGR_SMP_REASSEMBLY_BT
+	int ret;
+	bool started;
+
+	started = (zephyr_smp_reassembly_expected(&smp_bt_transport) >= 0);
+
+	LOG_DBG("started = %s, buf len = %d", started ? "true" : "false", len);
+	LOG_HEXDUMP_DBG(buf, len, "buf = ");
+
+	ret = zephyr_smp_reassembly_collect(&smp_bt_transport, buf, len);
+	LOG_DBG("collect = %d", ret);
+
+	/*
+	 * Collection can fail only due to failing to allocate memory or by receiving
+	 * more data than expected.
+	 */
+	if (!started && ret >= 0) {
+		/*
+		 * Transport context is attached to the buffer after first fragment
+		 * has been collected.
+		 */
+		struct smp_bt_user_data *ud =
+			(struct smp_bt_user_data *)zephyr_smp_reassembly_get_ud(&smp_bt_transport);
+
+		if (IS_ENABLED(CONFIG_MCUMGR_SMP_BT_LATENCY_CONTROL)) {
+			enable_low_latency(conn);
+		}
+
+		ud->conn = bt_conn_ref(conn);
+
+	} else if (ret == -ENOMEM) {
+		/* Failed to collect the buffer */
+		ret = BT_GATT_ERR(BT_ATT_ERR_INSUFFICIENT_RESOURCES);
+	} else if (ret < 0 && ret != -ENOMEM) {
```

Comment:
```suggestion
	} else if (ret < 0) {
```
ret == -ENOMEM is just the case above

Reply:
I am not sure how a compiler will order the conditionals and the `ret < 0` also catches `-ENOMEM`

Reply:
Done